### DISCUSSION TITLE
SSR - Text content did not match. Server - new Date()

### DISCUSSION THREAD

**Discussion by DevJett** [vercel/next.js#46958] [Help]
### Summary

I have the following SSR component contain current date that results in showing this error `Text content did not match. Server`

Clearly, the content of the page includes the output of new Date(), which changes every second. This means that the server-side rendered markup will be different from the client-side rendered content, resulting in the error `Text content did not match. Server`. 

Any ideas on how to solve such an error?

```
function SSRComponent() {

  return <p>The current time is {new Date().toString()}.</p>;
}

export async function getServerSideProps() {
  return { props: {} };
}
```

**Comment by icyJoseph** [comment]
For this case in particular, you can use `supressHydrationwarning`, like this: 

https://reactjs.org/docs/dom-elements.html#suppresshydrationwarning

```jsx
function SSRComponent() {

  return <p supressHydrationWarning>The current time is {new Date().toString()}.</p>;
}

export async function getServerSideProps() {
  return { props: {} };
}
```

- https://www.joshwcomeau.com/react/the-perils-of-rehydration/

In this article, there's a section on "Two-pass-rendering", which is the idea of using a `use-effect` on the client, to update the divergent data.

Although in some cases, like yours, the React docs say:

> However, in some rare cases, it is very hard or impossible to guarantee an exact match. For example, timestamps are expected to differ on the server and on the client.
